views/scan.ejs

```diff
@@ -110,61 +110,73 @@
-                        <select class="form-control" readonly>
-                          <option>=</option>
-                        </select>
-                      </td>
-                      <td>
-                        <input
-                          type="text"
-                          class="form-control"
-                          placeholder="Value"
-                          name="<%= filterKey %>"
-                          value="<%= filters[filterKey] %>"
-                        />
-                      </td>
-                    </tr>
+              <select name="queryableSelection"<% if (!Table.GlobalSecondaryIndexes) { %> disabled<% } %>>
+                <option value="table"
+                  <% Table.KeySchema.forEach(keySchemaItem => { %>
+                    data-key-<%= keySchemaItem.KeyType.toLowerCase() %>="<%= keySchemaItem.AttributeName %>"
+                  <% }) %>
+                  <% if (query.queryableSelection === 'table') { %> selected<% } %>
+                >[Table] <%= Table.TableName %></option>
+                <% if (Table.GlobalSecondaryIndexes) { %>
+                  <% for (const index of Table.GlobalSecondaryIndexes) { %>
+                    <% const attributes = index.KeySchema.map((keySchemaItem) => keySchemaItem.AttributeName) %>
+                    <option value="<%= index.IndexName %>"
+                      <% index.KeySchema.forEach(keySchemaItem => { %>
+                        data-key-<%= keySchemaItem.KeyType.toLowerCase() %>="<%= keySchemaItem.AttributeName %>"
+                      <% }) %>
+                      <% if (query.queryableSelection === index.IndexName) { %> selected<% } %>
+                      >[Index] <%= index.IndexName %>: <%= attributes.join(', ') %>
+                    </option>
                   <% } %>
+                <% } %>
+              </select>
+            </td>
+          </tr>
+          <tr>
+            <th>
+              Filter
+              <div style="margin-top: 1em">
+                <button id="add-filter" class="btn btn-secondary">Add Filter</button>
+              </div>
+            </th>
+            <td>
+              <table id="filters-table" class="table table-bordered">
+                <tbody>
                   <tr>
-                    <td style="width: 95px;"></td>
+                    <td style="width: 95px;">
+                      <button type="button" onclick="removeFilter(this)" class="filter-remove btn btn-danger">Remove</button>
+                    </td>
                     <td style="width: 300px">
                       <input
-                        data-filter-key
                         type="text"
-                        class="form-control"
-                        id="key"
+                        class="form-control filter-row-key"
                         placeholder="Key"
                         list="attributes"
                       />
                     </td>
                     <td>
```

Comment:
Can we specify some "big enough" width for attribute type and operator columns? They stretch too much and there is no need since content is static.

Reply:
Done